src/components/VCSMenu.tsx

```diff
@@ -80,22 +87,115 @@ export class VCSMenu extends React.Component<VCSMenuProps, VCSMenuState> {
+        selected_gm_group: gm_data[1]
+      });
+      this.graphicsMenuRef.setState({
+        selectedGroup: gm_data[0],
+        selectedMethod: gm_data[1],
+        tempGroup: gm_data[0]
+      });
+      console.log("Graphics selection loaded.");
+    } else {
+      // No meta data means fresh notebook, reset the graphics
+      this.graphicsMenuRef.resetGraphicsState();
+    }
+  }
+  async copyGraphicsMethod(
+    groupName: string,
+    methodName: string,
+    newName: string
+  ) {
+    try {
+      //Check that the method doesn't already exist in the selected group
+      if (this.props.getGraphicsList()[groupName].indexOf(newName) < 0) {
+        // If no duplicates, create command injection string
+        let command: string = `${newName}_${groupName} = `;
+        command += `vcs.create${groupName}('${newName}',source='${methodName}')`;
+        // Attempt code injection
+        await this.props.inject(command).then(async () => {
+          this.props.refreshGraphicsList();
+          // If successful, update the current state
+          await this.setState({
+            selected_gm_group: groupName,
+            selected_gm: newName
+          });
+          // Save selected graphics method to meta data
+          notebook_utils.setMetaData(
+            this.state.notebook_panel,
+            GRAPHICS_METHOD_KEY,
+            [this.state.selected_gm_group, this.state.selected_gm]
+          );
+        });
+        console.log(`Graphics method created: ${newName}`);
+      } else {
+        throw new Error("There is already a graphic method with that name.");
+      }
+    } catch (error) {
+      throw error;
+    }
+  }
+
   /**
    * @description inject code into the notebook loading the graphics method selected by the user
    * @param group the group name that the selected GM came from
    * @param name the specific GM from the group
    */
-  updateGraphicsOptions(group: string, name: string) {
-    this.setState({
-      selected_gm_group: group,
-      selected_gm: `${group}_${name}`
-    });
-    let gm_string = `${group}_${name} = vcs.get${group}('${name}')`;
-    this.props.inject(gm_string);
+  async updateGraphicsOptions(group: string, name: string): Promise<void> {
+    let gm_string: string = "";
+    if (name.indexOf(group) < 0) {
+      gm_string = `${name}_${group} = vcs.get${group}('${name}')`;
+    } else {
+      gm_string = `${name} = vcs.get${group}('${name}')`;
+    }
+
+    // Attempt code injection
+    try {
+      await this.props.inject(gm_string).then(() => {
+        // If successful, update the state
+        this.setState({
+          selected_gm_group: group,
+          selected_gm: name
+        });
+        // Save selected graphics method to meta data
+        notebook_utils.setMetaData(
+          this.state.notebook_panel,
+          GRAPHICS_METHOD_KEY,
+          [this.state.selected_gm_group, this.state.selected_gm]
+        );
+      });
+    } catch (error) {
```

Comment:
if you're just throwing the error and not handling it at all here, there's no reason to add the try/catch, it just adds extra code and a layer of brackets.